### ISSUE TITLE
[Armadillo] clapack installation order changes available features

### ISSUE THREAD

**Issue by arthur-tacca** [microsoft/vcpkg#5804]
Armadillo (added in #2954) has some optional functionality that depends on LAPACK being installed. If you use vcpkg to install clapack first and armadillo second then this functionality is enabled and works correctly. But if you use vcpkg to install armadillo first and clapack second then this functionality is disabled. This goes against the usual spirit of vcpkg, where extra functionality is enabled with feature packages, which ensures the required dependencies are installed.

Worse than this, if you use vcpkg to install clapack, then armadillo, then remove clapack, then you will get linking errors if you attempt to use armadillo. This is because vcpkg does not realise that armadillo has created an implicit dependency on clapack.

The detection of LAPACK and BLAS happens when Armadillo is configured (in its `CMakeLists.txt` and in files in `cmake_aux/Modules`). It is indicated by trace like this in its `config-x64-windows.log`:

    -- Found OpenBLAS: D:/src/vcpkg/installed/x64-windows/lib/openblas.lib
    -- Found LAPACK: D:/src/vcpkg/installed/x64-windows/lib/lapack.lib
    --      MKL_FOUND = NO
    --   ACMLMP_FOUND = NO
    --     ACML_FOUND = NO
    -- OpenBLAS_FOUND = YES
    --    ATLAS_FOUND = NO
    --     BLAS_FOUND = NO
    --   LAPACK_FOUND = YES

The obvious solution is to make a feature package for the LAPACK/BLAS functionality, with a dependency on clapack/openblas. This would require the port to *disable* BLAS/LAPACK support if the feature is not selected, even if the libraries are available, to avoid ending up in the linking error situation. I think this would require patching the `CMakeLists.txt` so that `OpenBLAS_FOUND` and similar symbols are set to `NO`, or patching `armadillo_bits/config.hpp` (which comes from `armadillo_bits/config.hpp.cmake`) so that `ARMA_USE_BLAS` and `ARMA_USE_LAPACK` are always undefined if the feature package is not selected.

## Example case

Here is a program that illustrates this behaviour:

    #include <armadillo>
    #include <iostream>
    int main() { 
        std::cout << arma::solve(arma::mat{{1, 2}, {3, 4}}, arma::vec{5, 11}) << "\n"; 
    }

Here is the result after installing armadillo only (this error is expected when running without LAPACK):

    error: solve(): use of LAPACK must be enabled

Here is the result after installing armadillo followed by clapack:

    error: solve(): use of LAPACK must be enabled

Here is the result after installing clapack by followed armadillo:

    1.0000
    2.0000

Here is the build output after installing clapack, followed by installing armadillo, and finally uninstalling clapack:

    1>------ Build started: Project: zarmatest, Configuration: Debug x64 ------
    1>zarmatest.obj : error LNK2019: unresolved external symbol sgesv_ referenced in function "void __cdecl arma::lapack::gesv<double>(int *,int *,double *,int *,int *,double *,int *,int *)" (??$gesv@N@lapack@arma@@YAXPEAH0PEAN00100@Z)
    1>zarmatest.obj : error LNK2019: unresolved external symbol dgesv_ referenced in function "void __cdecl arma::lapack::gesv<double>(int *,int *,double *,int *,int *,double *,int *,int *)" (??$gesv@N@lapack@arma@@YAXPEAH0PEAN00100@Z)

... and so on, you get the idea ...

**Comment by Voskrese** [comment]
> The detection of LAPACK and BLAS happens when Armadillo is configured (in its CMakeLists.txt and in files in cmake_aux/Modules). It is indicated by trace like this in its config-x64-windows.log:
can try

edit port
add

```
vcpkg_configure_cmake(
    SOURCE_PATH ${SOURCE_PATH}
    PREFER_NINJA
    OPTIONS
        -DDETECT_HDF5=false
        -DBLAS_LIBRARY:FILEPATH=E:/tools/vcpkg/installed/x86-windows/lib/openblas.lib
        -DLAPACK_LIBRARY:FILEPATH=E:/tools/vcpkg/installed/x86-windows/lib/openblas.lib
)
```

**Comment by JackBoosY** [comment]
Hi @arthur-tacca, thanks for reporting this issue!
This issue has been fixed in #6371, please update vcpkg and rebuild armadillo.

Thanks.

**Comment by arthur-tacca** [comment]
@JackBoosY Great job, thanks!
